File: backend/src/main/java/org/sjtu/backend/service/OrderService.java
package org.sjtu.backend.service;

import org.sjtu.backend.entity.*;

import java.util.List;
import java.util.Map;

public interface OrderService {
    //这里默认如果recipient == empty的话，说明采用user默认的信息
    Order generateOrder(String username, String recipient, String phone, String address);

    List<Order> showAllOrderOfUser(String username);

    List<List<OrderItem>> showAllOrder();

    List<Order> showOrderOfUserByBook(String username, String name);

    List<Order> showOrderByBook(String name);

    List<OrderItem> getOneOrder(int OrderId);

    List<OrderItem> getLatestOrder(String username);

    List<Book> showBestSeller(int dateNum);

    List<User> showCustomers(int dateNum);

    List<Book> showCustomerBuy(String username, int dateNum);

    boolean queryOrder(String username, String recipient, String phone, String address);
}

File: my-app/src/services/orderService.js
import config from '../utils/config';
import {postRequest, postRequest_v2} from "../utils/ajax";
import {message} from "antd";



export const getOrders=(username,callback)=>{
    const data = {username: username};
    const url = `${config.apiUrl}/order/showOrder`;
    postRequest_v2(url, data, callback);
}

export const getOrderItem = (orderId, callback) => {
    const data = {orderId: orderId};
    const url = `${config.apiUrl}/order/getOneOrder`;
    postRequest_v2(url, data, callback);
}

export const getLatestOrder=(username, callback) => {
    const data = {username: username};
    const url = `${config.apiUrl}/order/getLatestOrder`;
    postRequest_v2(url, data, callback);
}

export const getBestSeller=(dateNum, callback) => {
    const data = {dateNum: dateNum};
    console.log("dateNum in orderService: ", data);
    const url = `${config.apiUrl}/order/showBestSeller`;
    postRequest_v2(url, data, callback);
}

export const getCustomers=(dateNum, callback) => {
    const data = {dateNum: dateNum};
    console.log("dateNum in customer: ", data);
    const url = `${config.apiUrl}/order/showCustomers`;
    postRequest_v2(url, data, callback);
}

export const getCustomerBuy=(username, dateNum, callback) => {
    const data = {username:username, dateNum:dateNum};
    console.log("parameter in customer: ", data);
    const url = `${config.apiUrl}/order/showCustomerBuy`;
    postRequest_v2(url, data, callback);
}

export const getOrderResult=(user, callback) => {
    const data = {username: user.name, recipient: user.name, phone: user.phone, address: user.address};
    const url = `${config.apiUrl}/order/queryOrder`;
    postRequest_v2(url, data, callback);
}
